homeassistant/components/elkm1/config_flow.py

```diff
@@ -279,7 +281,7 @@ async def async_step_discovered_connection(self, user_input=None):
             else:
                 user_input[CONF_PREFIX] = ""
             errors, result = await self._async_create_or_error(user_input, False)
-            if not errors:
+            if result is not None:
```

Comment:
I checked all the code paths in `_async_create_or_error`, switching this check from error to result is cool, AFAICT.